src/hermes/model/merge.py

```diff
@@ -100,7 +100,10 @@ def __call__(self, target, path, value, **kwargs):
                 match target[key]:
                     case dict() as item: item.update(value)
                     case list() as item: item[:] = value
-                    case _: target[key] = value
+                    case _:
+                        if key in target:
```

Comment:
`if not key in target`, the access in line 100 should already throw, so, this check is superfluous.